fsspec/implementations/memory.py

```diff
@@ -280,6 +282,10 @@ def __init__(self, fs=None, path=None, data=None):
     def size(self):
         return self.getbuffer().nbytes
 
+    @property
+    def name(self):
+        return self.path.split("/")[-1]
```

Comment:
Hi Martin, 

I made the change. Just curious, why use rsplit over split?

Reply:
rsplit with `maxsplit=1`, will split of the last match value instead of splitting the entire string into lots of small strings and only using the last bit.

Reply:
It's also faster :)
```
In [8]: path = "/path/within/nested/directory/file"

In [9]: %timeit path.split("/")[-1]
164 ns ± 0.192 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [10]: %timeit path.rsplit("/", 1)[-1]
105 ns ± 0.0183 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
```